File: Python scripts/clean_logs.py
import os
import time

# Exact path to your logs folder
folder = r"C:\Users\Baha\Desktop\Developpement full stack\Lab 6\logs"

# Time in seconds for 7 days
seven_days = 7 * 24 * 60 * 60
now = time.time()

# Check each file in the folder
for filename in os.listdir(folder):
    file_path = os.path.join(folder, filename)
    if os.path.isfile(file_path) and file_path.endswith(".log"):
        age = now - os.path.getmtime(file_path)
        if age > seven_days:
            print(f"[DELETE] {file_path}")
            os.remove(file_path)
        else:
            print(f"[KEEP] {file_path}")

File: Go scripts/clean_logs/clean_logs.go
package main

import (
	"fmt"
	"os"
	"path/filepath"
	"time"
)

func main() {
	// Exact path to your logs folder
	folder := `C:\Users\Baha\Desktop\Developpement full stack\Lab 6\logs`

	now := time.Now()

	err := filepath.Walk(folder, func(path string, info os.FileInfo, err error) error {
		if err != nil {
			fmt.Println("Error:", err)
			return nil
		}
		// Only files with .log extension
		if !info.IsDir() && filepath.Ext(path) == ".log" {
			age := now.Sub(info.ModTime()).Hours() / 24 // Age in days
			if age > 7 {
				fmt.Println("[DELETE]", path)
				os.Remove(path)
			} else {
				fmt.Println("[KEEP]", path)
			}
		}
		return nil
	})

	if err != nil {
		fmt.Println("Error walking the path:", err)
	}
}
